helix-core/src/main/java/org/apache/helix/task/TaskStateModel.java

```diff
@@ -283,6 +294,63 @@ public void reset() {
     }
   }
 
+  /**
+   * Loads Task and TaskFactory classes for command input from
+   * a JAR file, and register the TaskFactory in _taskFactoryRegistry.
+   * @param command The command indicating what task to be loaded
+   */
+  private void loadNewTask(String command) {
+    // Read ZNRecord containing task definition information.
+    ZNRecord taskConfig = _manager.getHelixDataAccessor().getBaseDataAccessor()
+        .get("/" + _manager.getClusterName() + "/" + TASK_PATH + "_" + command,null, 0);
+    if(taskConfig == null) {
+      LOG.info("Failed to read ZNRecord for task " + command + " for instance " + _manager.getInstanceName()
+          + " in cluster " + _manager.getClusterName() + ".");
+      throw new IllegalStateException("No ZNRecord for task " + command);
+    }
+
+    // Open the JAR file containing Task(s) and TaskFactory classes.
+    File taskJar;
```

Comment:
Since loading the JAR from the local directory is not a permanent solution and rather not very useful, consider using Java Interface to make this pluggable. That's a better design principle and will allow you to extend your code more easily. The way it's hard-coded right now, it will be difficult to change the underlying implementation for the jar loader.